perfkitbenchmarker/virtual_machine.py

```diff
@@ -148,6 +163,17 @@ def __repr__(self):
   def __str__(self):
     return self.ip_address
 
+  def __getstate__(self):
+    """Get state, excluding _has_remote_command_script_lock."""
```

Comment:
`__getstate__` and `__setstate__` are used for pickling (Python serialization) objects. Locks can't be serialized, so these two methods:
* Drop the lock from the state to be serialized.
* Create a new lock when the object is deserialized.

Added some comments.